File: Arrays/ProblemSolutions/jumpGame.java
import java.util.*;

class Solution {
    public boolean canJump(int[] nums) {
        int lastIndex = nums.length-1;
        for(int i=nums.length-1; i>=0; i--){
            if(i+nums[i] >= lastIndex)
                lastIndex = i;
        }
        return lastIndex == 0;
    }
}

// Question: https://leetcode.com/problems/jump-game/?envType=study-plan-v2&envId=top-interview-150

File: Arrays/ProblemSolutions/jumpGame.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True
        
        target = len(nums) - 1
        canGo = 0

        for i in range(0, len(nums)):
            canGo = max(canGo, nums[i])
            if canGo >= 1:
                if canGo + i >= target:
                    return True
                canGo -= 1
            else:
                return False

# Question: https://leetcode.com/problems/jump-game/